src/Features/Core/Portable/InitializeParameter/AbstractInitializeParameterCodeRefactoringProvider.cs

```diff
@@ -122,6 +195,24 @@ private bool CanOfferRefactoring(SyntaxNode functionDeclaration, SemanticModel s
             return true;
         }
 
+        protected TParameterSyntax GetParameterNode(SyntaxToken token, int position)
```

Comment:
do you need this helper?  this should all fall out from using `TryGetRelevantNodeAsync<TParameterSyntax>()` right?

Reply:
As long as we update the document in the loop we will need this (otherwise we'll get another syntax tree). This can be replaced once document is only updated at the end.

Reply:
Will try to address this in a follow-up PR